src/polytool.ts

```diff
@@ -0,0 +1,48 @@
+'use strict';
+
+import * as commandLineCommands from 'command-line-commands';
+import {HelpCommand} from './commands/help';
+import {ServeCommand} from './commands/serve';
+import {Command} from './commands/command';
+
+export class Polytool {
+
+  commandDescriptors = [];
+  commands : Map<String, Command> = new Map();
+  cli : commandLineCommands.CLI;
+
+  constructor() {
+    this.addCommand(new HelpCommand(this.commands));
+    this.addCommand(new ServeCommand());
+  }
+
+  addCommand(command) {
+    this.commands.set(command.name, command);
```

Comment:
What is the intended behavior if a user calls addCommand with the same name twice?